src/ExpressReceiver.ts

```diff
@@ -218,7 +218,7 @@ async function verifyRequestSignature(
   body: string,
   signature: string,
   requestTimestamp: number): Promise<void> {
-  if (!signature || !requestTimestamp) {
+  if (!Boolean(signature) || !Boolean(requestTimestamp)) {
```

Comment:
ditto to previous comment.

Reply:
I think somewhere in your code you're relying on the implicit Boolean conversion of `NaN` for the request timestamp. Also does a different thing if you set it `!== undefined`, which I'm not sure is wanted here?

Reply:
You're converting the `x-slack-request-timestamp` header into a number here: https://github.com/slackapi/bolt/blob/master/src/ExpressReceiver.ts#L191

And then, in the spec, you're verifying that it detects that missing header: https://github.com/slackapi/bolt/blob/master/src/ExpressReceiver.spec.ts#L111

So if you compare `requestTimestamp !== undefined` it won't work, but maybe fine to use `requestTimestamp !== `NaN` since it's assumed the type of the input is a number?

Reply:
You're converting the `x-slack-request-timestamp` header into a number here: https://github.com/slackapi/bolt/blob/master/src/ExpressReceiver.ts#L191

And then, in the spec, you're verifying that it detects that missing header: https://github.com/slackapi/bolt/blob/master/src/ExpressReceiver.spec.ts#L111

So if you compare `requestTimestamp === undefined` it won't work, but maybe fine to use `isNaN(requestTimestamp)` since it's assumed the type of the input is a number? Updated to that.

Reply:
Should L190-L191 look like the following?

```
      const signature = req.headers['x-slack-signature'] as string | undefined;
      const ts = Number(req.headers['x-slack-request-timestamp']) as number | NaN;
```

Then the `verifyRequestSignature()` function's arguments would change type:

* `signature: string` -> `signature: string | undefined`
* `requestTimestamp: number` -> `requestTimestamp: number | NaN`

If we did this, the `verifyRequestSignature()` function would be more resilient in terms of how it can be used, especially if we end up refactoring it into another package as the comment above suggests.

However, I don't quite think `NaN` was ever intended to be an acceptable value for `requestTimestamp`. That just feels like a mistake on L191. I think the intention would be for this argument's type to be `number | undefined`. I'd personally rather see this change. It would also be more resilient than today's implementation.

Reply:
`NaN` is a value (not a type - `typeof NaN === 'number'`), so  I don't think `as number | NaN` would work for L191 ?

My thinking around checking if it's `NaN` - `isNaN(undefined)` is `false` anyway. Agree `number | undefined` is probably more the intention/semantically what you were originally trying to handle, but `isNaN` handles that case plus the case that someone sends something like "Hey there!" as the timestamp, so the timestamp is defined but in trying to convert the string to a number you get `NaN`, which this would catch. 

I'll take a stab at this - I think the thing that makes sense is to have the function that verifies the request signature get the header as is and do the number conversion/check, since there's a couple things you want to check for there. I can take a stab at this and see how it looks.

Reply:
Ended up just thinking it's probably better to check for the two cases explicitly - it's undefined, or it's invalid in a couple of different ways (not a number, too old). So just broke it out into two checks.